File: src/test/java/github/ptrteixeira/presenter/MockWebScraper.java
package github.ptrteixeira.presenter;

import github.ptrteixeira.model.ConnectionFailureException;
import github.ptrteixeira.model.Match;
import github.ptrteixeira.model.Standing;
import github.ptrteixeira.model.WebScraper;
import javafx.collections.ObservableList;

import java.util.List;

/**
 * Created by pteixeira on 6/6/16.
 */
public class MockWebScraper implements WebScraper {
  @Override
  public ObservableList<Standing> getStandings(String sport) throws ConnectionFailureException {
    return null;
  }

  @Override
  public ObservableList<Match> getSchedule(String sport) throws ConnectionFailureException {
    return null;
  }

  @Override
  public List<String> getSelectableSports() {
    return null;
  }

  @Override
  public void clearCache(String sport) {

  }
}

File: src/test/kotlin/com/github/ptrteixeira/nusports/presenter/MockWebScraper.kt
/* Released under the MIT license, 2018 */

package com.github.ptrteixeira.nusports.presenter

import com.github.ptrteixeira.nusports.model.Match
import com.github.ptrteixeira.nusports.model.Standing
import com.github.ptrteixeira.nusports.model.WebScraper

class MockWebScraper(private val mockImpl: SyncWebScraper) : WebScraper {
    override val selectableSports: List<String>
        get() = mockImpl.selectableSports()

    override suspend fun getStandings(sport: String): List<Standing> = mockImpl.getStandings(sport)

    override suspend fun getSchedule(sport: String): List<Match> = mockImpl.getSchedule(sport)

    override fun clearCache(sport: String) {
        mockImpl.clearCache(sport)
    }
}
